### DISCUSSION TITLE
Why is free-tier that slow?

### DISCUSSION THREAD

**Discussion by 8byr0** [supabase/supabase#4862] [Questions]
Hi there!
I'm exploring supabase and worry a bit about performances.

I created a free-tier app (no content yet) and tested this simple query (I started with the query and used EXPLAIN ANALYZE after noticing it was slow) :
```sql
EXPLAIN ANALYZE (
  SELECT * from
  (
    SELECT count(pid) as active_connections FROM pg_stat_activity WHERE state = 'active'
  ) active_connections,
  (
    SELECT setting as max_connections from pg_settings where name = 'max_connections'
  ) max_connections
);
``` 
The results are quite bad:
![image](https://user-images.githubusercontent.com/21955112/148564019-a934a555-b54b-41b1-bffe-aeb8274a03b2.png)
1.6s for a query that only reads db data makes me feel sceptical...

I also tested out slack clone (https://github.com/supabase/supabase/tree/master/examples/nextjs-slack-clone) and it was quite unusable (so slow response time, messages not showing instantly when sent, same after channel creation).

Are there any known limitations that could affect performance like this? Is it due to free tier?

**Comment by steve-chavez** [comment]
> 1.6s for a query that only reads db data makes me feel sceptical...

Hm, your image shows 1.6 **milliseconds**, not seconds.

**Reply by 8byr0** [reply]
Are you sure ? I guess . is thousands separator because it really feels like 1.6 seconds :

https://user-images.githubusercontent.com/21955112/148593107-41b59cdf-b47e-4a81-846f-a663b789b1bd.mp4

**Reply by 8byr0** [reply]
(I tried with both wifi connection and LTE and result is always the same)

**Reply by GaryAustin1** [reply]
The actual query was 1.6 msec as Steve said.  Your video is measuring all the round trip times to the server to run the sql window, send the sql command, get back a result.  Look at the network tab and you will see several round trips.   You really need to run using the API to get an idea on response time as those send the request and get response in one network request, but the Postgres database is only using 1.6 msec of that.
Edit.  Just try running  "show search_path", and see how long the UI SQL window takes to do all the stuff it does for a just to get a user setting.

**Reply by 8byr0** [reply]
You're right, I should have started with that. TTFB is almost 3s for the query:
![image](https://user-images.githubusercontent.com/21955112/148601493-05b9f4bc-5fec-4aeb-8ef0-26f9b99c75a4.png)

Seems like middlewares before/after SQL query take many more time than expected

**Reply by 8byr0** [reply]
I'll give a try to API tomorrow to see if it makes a difference.

I also checked request on slack clone and here are the results for refetch query after sending a message :  
![image](https://user-images.githubusercontent.com/21955112/148602041-735b538d-031c-4122-acdc-4e3baf905377.png)

It was far slower earlier this day, how likely is it that number of active realtime connections affect response time?

**Reply by 8byr0** [reply]
@GaryAustin1  when running `show search_path` here are the results:
![image](https://user-images.githubusercontent.com/21955112/148602424-81235226-a598-4951-a8a7-53988df5ffe2.png)

![image](https://user-images.githubusercontent.com/21955112/148602376-cabaaf8c-3247-4f85-b799-5a9cad1c3e14.png)

**Reply by GaryAustin1** [reply]
If that is from the sql window in the UI it is not reflective of the API, it is doing lots of things including multiple network requests.
I normally see around 200ms for the entire network time for a simple supubase js select call.  Although there is also overhead for https preflight.

**Reply by 8byr0** [reply]
I'll perform tests with API to check how different it is. 
Do you have any idea why TTFB is such high using UI?

**Reply by GaryAustin1** [reply]
No but it is doing lots of different calls and they are slow on my UI also. 
 
Edit: In thinking about it, the UI is running as a separate server, that is in turn making calls to your SB instance.  So besides the number of calls being made to the UI, it is having to go out with more delay to your instance's db.

The api using the same instance runs  200ms for a select.  With preflight request for https, it would be about 400ms total from function call to response.
